src/sdam/topology.ts

```diff
@@ -178,8 +178,11 @@ export interface SelectServerOptions {
   session?: ClientSession;
   operationName: string;
   previousServer?: ServerDescription;
-  /** @internal*/
-  timeout?: Timeout;
+  /**
+   * @internal
+   * TODO(NODE-5685): Make this required
+   * */
+  timeoutContext?: TimeoutContext;
```

Comment:
Change streams didn't end up using LegacyTimeoutContext so this is still optional

Reply:
To clarify, should it be optional? Or should it be required?

Reply:
The desire is for it to be required from a code maintenance perspective ideally we always have a context (either CSOT or not) passed this point so code doesn't have to think about the two separate cases. - this does not have API impact so we can improve this later.